composite/apy-finance/src/registry/index.ts

```diff
@@ -0,0 +1,40 @@
+import { ethers } from 'ethers'
+import registryAbi from './IRegistry.json'
+import assetAllocationAbi from './IAssetAllocation.json'
+
+const getRegistry = (address: string, rpcUrl: string, abi: any): ethers.Contract => {
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
+  return new ethers.Contract(address, abi, provider)
+}
+
+type Allocations = {
+  components: string[]
+  units: string[]
+}
```

Comment:
This type is a relic of `defi-pulse` underlying contract implementation and is inflexible. How about we change this type to something more usable like:

```javascript
type TokenAllocation = {
   symbol: string
   decimals: number
   balance: string | BN
}
```

This type should then be moved to the `token-allocations` adapter and used by APY & DeFi Pulse.

Reply:
Yep agree